packages/suite/src/utils/wallet/transactionUtils.ts

```diff
@@ -367,3 +368,282 @@ export const enhanceTransaction = (
+    }
+
+    switch (operator) {
+        case '<':
+            return bnTargetAmount.lte(amount);
+        case '>':
+            return bnTargetAmount.gte(amount);
+        case '=':
+            return bnTargetAmount.eq(amount);
+        case '!=':
+            return !bnTargetAmount.eq(amount);
+        // no default
+    }
+};
+
+const searchDateRegex = new RegExp(/^[0-9]{4}-[0-9]{2}-[0-9]{2}$/);
+export const simpleSearchTransactions = (
+    transactions: WalletAccountTransaction[],
+    accountMetadata: AccountMetadata,
+    search: string,
+) => {
+    // Trim
+    search = search.trim();
+
+    // If the string is empty or only contains search operators, there's no search
+    if (['', ...searchOperators].includes(search)) {
+        return transactions;
+    }
+
+    // Check for date
+    if (searchDateRegex.test(search)) {
+        // Add search operator so it gets picked up below
+        search = `=${search}`;
+    }
+
+    // If it's an amount search (starting with <, > or = operator)
+    const searchOperator = searchOperators.find(k => search.startsWith(k));
+    if (searchOperator) {
+        // Remove search operator from search string
+        search = search.replace(searchOperator, '').trim();
+
+        // Is date?
+        if (searchDateRegex.test(search)) {
+            const timestamp = +new Date(`${search}T00:00:00Z`) / 1000;
+            switch (searchOperator) {
+                case '>':
+                    return transactions.filter(t => t.blockTime && t.blockTime > timestamp);
+                case '<':
+                    return transactions.filter(
+                        t => t.blockTime && t.blockTime < timestamp + 24 * 60 * 60,
+                    );
+                case '=':
+                    return transactions.filter(
+                        t =>
+                            t.blockTime &&
+                            t.blockTime > timestamp &&
+                            t.blockTime < timestamp + 24 * 60 * 60,
+                    );
+                case '!=':
+                    return transactions.filter(
+                        t =>
+                            t.blockTime &&
+                            (t.blockTime < timestamp || t.blockTime > timestamp + 24 * 60 * 60),
+                    );
+                // no default
+            }
+        }
+
+        // Is number?
+        if (!Number.isNaN(search)) {
+            const amount = new BigNumber(search);
+            return transactions.filter(t => numberSearchFilter(t, amount, searchOperator));
+        }
+
+        return [];
+    }
+
+    const txsToSearch: string[] = [];
+
+    // Searching for an amount (without operator)
+    if (!Number.isNaN(search)) {
+        const foundTxsForNumber = transactions.flatMap(t => {
+            const targetAmount = getTargetAmount(t.targets[0], t);
+            if (!targetAmount || !targetAmount.includes(search)) {
+                return [];
+            }
+            return t.txid;
```

Comment:
```suggestion
            return [t.txid];
```

 is it working when you return an object?

Reply:
Returning a string with `flatMap` works. Basically:
```javaScript
['a', 'b', 'c'].flatMap(l => {
	if (l === 'a') {
		return [];
	}
	return l; // Return string
}); // ['b', 'c']
```

is the same as:
```javaScript
['a', 'b', 'c'].flatMap(l => {
	if (l === 'a') {
		return [];
	}
	return [l]; // Return array
}); // ['b', 'c']
```

Reply:
TIL :)